### PR TITLE
QGIS missing dependencies 2 of 4: Qwt and QwtPolar

### PR DESCRIPTION
**Pull request by ceholden**
PR adds builds for Qwt and QwtPolar -- two plotting toolkits based on Qt that are dependencies for the real goal, adding [QGIS](qgis.org) to conda-forge (which I have working Linux, locally on Arch and on our CentOS cluster). 

There are 5 recipes I've written up in total today. Having not added anything to conda-forge yet, I've started on these two but can pursue QGIS soon after the dependencies are merged. 

I'll briefly describe the path to getting QGIS added before getting back to this PR:

### Path to QGIS
Conda's defaults and conda-forge cover every dependency I'm used to including in [QGIS builds](https://github.com/qgis/QGIS/blob/master/INSTALL) except for:

* [ ] `qwt`
* [ ] `qwtpolar`
* [ ] `qca`
    * Next recipe to work on; [already have a branch ready to submit](https://github.com/ceholden/staged-recipes/tree/qca)
* [ ] `qscintilla2`
    * I have a temporary recipe copied from conda/conda-recipes#624 which can be replaced whenever the PR is merged

I have a [branch with the QGIS recipe](https://github.com/ceholden/staged-recipes/tree/qgis) that I've built on 2 Linux distros already, but I had to hack in a patch so that QGIS would recognize the Qt plugins (`plugins/sqldrivers/qsqlite.so`) after being packaged by conda. Happy to discuss what to do with QGIS later in this PR or in another (like using `binary_has_prefix_files`). For now, I have an up-to-date copy of QGIS that I can install on the various computing clusters I use without compiling each time, so I'm really :smile: 

## Qwt and QwtPolar

If I'm not mistaken, both libraries require patching to allow us to install them within conda. I'm assuming the Mac build can work without significant modification, but I have no idea what I'm doing on Windows. These are dependencies for QGIS first and foremost and I had a hard enough time just getting QGIS working on Linux, so if anyone would like to contribute on the OSX and Windows side I would be very happy. I can work toward the OSX build myself if need be, but I'm out of touch with that platform. 

Please edit, critique, comment, etc. as need be. I bundled the two recipes together because they're so related, but I'm fine submitting separate PRs.

### CODE REVIEW

File: recipes/qwtpolar/build.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+cd build/
+
+export QWT_POLAR_INSTALL_PREFIX=$PREFIX
+export QT_POLAR_INSTALL_PREFIX=$PREFIX
+
+$BIN/qmake ../qwtpolar.pro
+
+CPPFLAGS="-I$PREFIX/include" LDFLAGS="-L$PREFIX/lib" \
+make
+
```
Comment (msarahan) [inline recipes/qwtpolar/build.sh]: is there any meaningful make check test here?

File: recipes/qwt/build.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+BIN=$PREFIX/bin
+
+[[ -d build ]] || mkdir build
+cd build
+
+QWT_INSTALL_PREFIX=$PREFIX $BIN/qmake ../qwt.pro
+make
+make install
```
Comment (msarahan) [inline recipes/qwt/build.sh]: is there any meaningful make check/test here?

**Comment by msarahan** [comment]
Nice work.  If there are test suites that are runnable with make, or otherwise, please run them.  Otherwise, this PR LGTM.

**Comment by ceholden** [comment]
@msarahan Thank you for reviewing my PR and apologies for the delayed reply.

Neither library has a test suite, but they do provide examples to build. I've modified the "build.sh" scripts to build these examples as a check to ensure other packages can link against libqwt and libqwtpolar.

**Comment by ceholden** [comment]
My addition failed because I was testing for existence of one of the example binaries. On Linux, this binary is `bin/animation` but on OSX it's `bin/animation.app/Contents/MacOS/animation` and I didn't switch the test by the OS.

If the examples link correctly, then that should be proof enough that `libqwt` and `libqwtpolar` are functional and I can avoid checking for the actual binaries.

**Comment by ocefpaf** [comment]
@ceholden I missed this one. Do you mind rebasing against latest master to "modernize" the PR so we can re-run with the latest changes?

File: recipes/qwt/build.sh [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+BIN=$PREFIX/bin
+
+[[ -d build ]] || mkdir build
+cd build
+
+QWT_INSTALL_PREFIX=$PREFIX $BIN/qmake ../qwt.pro
+
+make
+make install
```
Comment (ocefpaf) [inline recipes/qwt/build.sh]: Is there a `make check`?

**Comment by ceholden** [comment]
Thanks for your attention, @ocefpaf. I finally got around to finishing this up. The switch from `qmake` to `qmake-qt4` killed a bit of momentum, but the real bear was altering the patching of the qmake project files to put everything in the right places.

I'm punting on the Windows build at least until I can everything including QGIS merged for Linux/Mac, so I think we can ignore the Appveyor failure. Looks like some temporary network issue caused it for one entry in the build matrix.

**Comment by ocefpaf** [comment]
@ceholden I have to confess those are quite the complex build and I am the best guy to review them. With that said I am OK merging this and addressing any future problem in the feedstock.

**Pull request merged**